### ISSUE TITLE
Here is a bug on linear loss computation

### ISSUE THREAD

**Issue by begeekmyfriend** [Rayhane-mamah/Tacotron-2#113]
In the [expression](https://github.com/Rayhane-mamah/Tacotron-2/blob/master/tacotron/models/tacotron.py#L220) computing linear loss, `num_mels` should have been `num_freq`. See Keith Ito's [version](https://github.com/keithito/tacotron/blob/master/models/tacotron.py#L118-L119). It seems that this model does not compute the loss including effective bandwidth of the audio.

**Comment by Yeongtae** [comment]
@begeekmyfriend how to fix it? 
When we fix it, what is the improvement compare with the previous one?

**Comment by begeekmyfriend** [comment]
> linear_loss = 0.5 * tf.reduce_mean(l1) + 0.5 * tf.reduce_mean(l1[:,:,0:n_priority_freq])

This expression means we use 0.5 weight of the whole bandwidth of the frequency plus the remaining 0.5 weight of the priority bandwidth of the frequency as the complete linear loss to train the model. The wrong `num_freq` factor would make influence on the bandwidth. So in my humble opinion, the higher frequency of the targets would be taken part in fitting the ground truth audio.

**Comment by Yeongtae** [comment]
@begeekmyfriend Thank you for your good opinion.
did you test it? did it reduce noise the result such as 'step-xxxx-wave-from-mels'.

**Comment by begeekmyfriend** [comment]
It does nothing with the mel outputs. By the way, the linear outputs typically perform better than mel ones.

**Comment by Yeongtae** [comment]
![image](https://user-images.githubusercontent.com/13597229/43190502-2a9ea042-9034-11e8-9192-c5abb25fe92d.png)
@begeekmyfriend does it affect this part?
Thanks a lot.

**Comment by begeekmyfriend** [comment]
It definitely does because I have expand both the fft size and the number of frequency of linear outputs. So the audio signal process would be affected. By the way, these hyper parameters do not match wavenet vocoder but only for Griffin-Lim.

**Comment by begeekmyfriend** [comment]
We can also use L2 loss for linear outputs if you think L2 is better than L1
```python
n_priority_freq = int(4000 / (hp.sample_rate * 0.5) * hp.num_freq)
linear_loss = 0.5 * tf.losses.mean_squared_error(self.linear_targets, self.linear_outputs) \
        + 0.5 * tf.losses.mean_squared_error(self.linear_targets[:,:,0:n_priority_freq], self.linear_outputs[:,:,0:n_priority_freq])
```

**Comment by Yeongtae** [comment]
This is my test results.
Two models look like converge around loss 0.4.

--num_mels: 10000iteration--
![step-10000-eval-mel-spectrogram](https://user-images.githubusercontent.com/13597229/43235382-a1371a5e-90bb-11e8-918c-ff3e3246b601.png)
![step-10000-eval-align](https://user-images.githubusercontent.com/13597229/43235383-a28739ca-90bb-11e8-8e24-405a92ff6c7d.png)

--num_freq: 10000iteration--
![step-10000-eval-mel-spectrogram](https://user-images.githubusercontent.com/13597229/43235551-e51e30cc-90bb-11e8-90d1-40f56e5deb03.png)
![step-10000-eval-align](https://user-images.githubusercontent.com/13597229/43235552-e542d198-90bb-11e8-889a-4439b8a67b6c.png)

Because of the experimental results, it seems to not good choice to apply num_freq in linear loss.

**Comment by begeekmyfriend** [comment]
I am afraid there are problems in your audio dataset. In my test it would achieve convergence in 4K steps.
![step-4000-align](https://user-images.githubusercontent.com/6031938/43236654-e977229e-90b8-11e8-99b1-4f7f1ce798de.png)

**Comment by begeekmyfriend** [comment]
I am afraid there might be problems in your dataset. In my test it would achieve convergence in 4K steps when I adopted the solution mentioned on the 8th floor that used MSE for linear loss.
![step-4000-align](https://user-images.githubusercontent.com/6031938/43236715-30632e14-90b9-11e8-8085-982b26680da9.png)

**Comment by Starlon87** [comment]
@begeekmyfriend 为什么你的4000 steps，loss = 0.59就可以看起来如此收敛，而我这边70000 steps，loss = 0.37还没苗条的曲线?...

![step-70000-eval-align](https://user-images.githubusercontent.com/38202890/43244225-92f2b176-90dc-11e8-823e-ad0e8b49ebb8.png)

**Comment by v-yunbin** [comment]
@Starlon87 your amount of data is  different from his.

**Comment by Rayhane-mamah** [comment]
@begeekmyfriend my good friend you are correct once more!

I have fixed that.. I apologize for the typo :) Thanks for your feedback!
